website/docs/getting-started/basics.md

```diff
@@ -8,15 +8,34 @@ odo abstracts Kubernetes concepts into developer focussed terminology. In this d
 
 ### Concepts
 
-#### Component
+* Component
+  A component is a microservice. Each component can be run and deployed separately. Examples of components: Warehouse API Backend, Inventory API, Web Frontend, Payment Backend
+  > A component in odo is an equivalent of a deployment or deploymentconfig in Kubernetes. Creating a new component in odo will create a deployment in Kubernetes or for git and binary based component creation, it will create deploymentconfig.
 * Application
+  An application consists of multiple components which may span over multiple projects, and work individually to build the entire application. Examples of applications: e-Shop, Hotel Reservation System, Online Booking
+  > An application can be considered as an equivalent of labels in Kubernetes that help in grouping a set of resources.
 * Project
+  A project is your source code, tests, and libraries organized in a separate single unit.
+  > A project in odo is an equivalent of a namespace in Kubernetes. Creating a project in odo will create a namespace in Kubernetes with the same name.
 * Context
+  A context is the directory where the source code, tests, and libraries for your component resides. A single context can only contain a single component.
 * URL
+  A URL exposes your component to be accessed outside the cluster.
+  > A URL is an equivalent of a service in Kubernetes.
 * Storage
+  A storage volume is [PVC](https://kubernetes.io/docs/concepts/storage/volumes/#persistentvolumeclaim) which is a way for you to "claim" persistent storage without knowing the details of the environment. Storage volume can persist data across restarts and rebuilds of a component.
```

Comment:
Using k8s terminology to explain odo concept is not a good idea. It just means we fail to explain (or even understand ourselves!) the abstractions we created to make k8s terminology not required for the user.

Reply:
Again, picked it up from existing docs, and it seemed to make sense to me, but I get your point.